### ISSUE TITLE
Tutorial part 2, cannot bypass authentication

### ISSUE THREAD

**Issue by morenoh149** [tomchristie/django-rest-framework#4206]
## Checklist

- [x] I have verified that that issue exists against the `master` branch of Django REST framework.
- [x] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [x] This is not a usage question. (Those should be directed to the [discussion group](https://groups.google.com/forum/#!forum/django-rest-framework) instead.)
- [x] This cannot be dealt with as a third party library. (We prefer new functionality to be [in the form of third party libraries](http://www.django-rest-framework.org/topics/third-party-resources/#about-third-party-packages) where possible.)
- [x] I have reduced the issue to the simplest possible case.
- [x] I have included a failing test as a pull request. (If you are unable to do so we can still accept the issue.)

## Steps to reproduce
My repo at this commit https://github.com/morenoh149/djangoRestFrameworkTutorial/commit/480e116cb6c48096a4a1a774ad7361467599ed29

## Expected behavior
Working through the [tutorial - part 2](http://www.django-rest-framework.org/tutorial/2-requests-and-responses/) at the end of the section we should be able to get snippets in json and html format.

## Actual behavior
Instead I get
```http
▶ http http://127.0.0.1:8000/snippets.json  # JSON suffix          
HTTP/1.0 403 Forbidden
Allow: POST, OPTIONS, GET
Content-Type: application/json
Date: Mon, 20 Jun 2016 19:22:55 GMT
Server: WSGIServer/0.1 Python/2.7.10
Vary: Accept, Cookie
X-Frame-Options: SAMEORIGIN

{
    "detail": "Authentication credentials were not provided."
}
```

I thought adding `@csrf_exempt` would allow the endpoints to function as before but it did not change the outcome.

**Comment by tomchristie** [comment]
First thing I can see that's different is that you still have '@csrf_exempt' decorators in your codebase, which we don't in the tutorial at that point. Not sure if that's the issue but you don't want those there - @api_view will ensure you get the correct behaviour (CSRF exempt for session authenticated, but not for others)

If you can confirm you still see an issue when the codebase exactly matches the tutorial (perhaps run through from start to be sure) then could consider reopening.

**Comment by morenoh149** [comment]
Indeed. I'm going through the tutorial from the beginning and removed `@csrf_exempt` from the functions. I added afterwards attempting to solve the authentication issue. I can try again without the decorator in a couple of minutes.

**Comment by morenoh149** [comment]
The problem was that I did the quickstart before the tutorial. With the same repo. The quickstart has you set the default_permission_classes http://www.django-rest-framework.org/tutorial/quickstart/#settings which would require you to authenticate when you curl at http://www.django-rest-framework.org/tutorial/2-requests-and-responses/#hows-it-looking I've made this change and can continue with the tutorial.
